src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -571,17 +603,33 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
                 if (NetEventSource.IsEnabled) NetEventSource.ClientSendCompleted(this, response, request);
                 return response;
             }
+            catch (TaskCanceledException e) when (TimeoutFired(timeoutState, callerToken))
+            {
+                ThrowTimeoutException(ExceptionDispatchInfo.Capture(e));
+                throw;
+            }
             catch (Exception e)
             {
                 HandleFinishSendAsyncError(e, cts);
                 throw;
             }
             finally
             {
-                HandleFinishSendAsyncCleanup(cts, disposeCts);
+                HandleFinishSendAsyncCleanup(cts, disposeCts, timer);
             }
         }
 
+        private bool TimeoutFired(TimeoutState timeoutState, CancellationToken callerToken)
+            => !callerToken.IsCancellationRequested && !_pendingRequestsCts.IsCancellationRequested && timeoutState != null && timeoutState.IsFired;
+
+        private static void ThrowTimeoutException(ExceptionDispatchInfo exceptionInfo)
```

Comment:
Why are you using ```ExceptionDispatchInfo``` if you're not throwing the exception with its Throw method? AFAIK putting exception into InnerException and throwing the new one will not change the inner's exception call stack.

Reply:
It's a leftover from some preliminary version, not needed anymore. I will remove it.